services/src/main/java/uk/gov/moj/sdt/services/messaging/asb/MessageSender.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.moj.sdt.services.messaging.asb;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusMessage;
+import com.azure.messaging.servicebus.ServiceBusSenderClient;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.moj.sdt.services.messaging.api.ISdtMessage;
+
+@Slf4j
+@Component
+public class MessageSender {
+
+    private String connectionString;
+
+    private String queueName;
+
+    private JsonConverter jsonConverter;
+
+    @Autowired
+    public MessageSender(@Value("${jms.servicebus.internal.queues.inbound.connection-string}") String connectionString,
+                         @Value("${jms.servicebus.internal.queues.inbound.queue-name}") String queueName,
+                         JsonConverter jsonConverter) {
+        this.connectionString = connectionString;
+        this.queueName = queueName;
+        this.jsonConverter = jsonConverter;
+    }
+
+    public void sendMessage(String queueName, ISdtMessage sdtMessage) {
+        try {
+            final ServiceBusSenderClient senderClient = new ServiceBusClientBuilder()
+                .connectionString(connectionString)
+                .sender()
+                .queueName(queueName)
+                .buildClient();
+            log.debug("Connected to queue {}", queueName);
+            String message = jsonConverter.convertToString(sdtMessage);
+            ServiceBusMessage serviceBusMessage = new ServiceBusMessage(message);
+            senderClient.sendMessage(serviceBusMessage);
+
+            log.debug("Message has been sent to the queue {}", queueName);
+        } catch (Exception e) {
+            log.error("Error while sending the message to queue:{}", e.getMessage());
```

Comment:
The original exception handling included the SDT request reference in the message being logged (see lines 122 and 123 in previous version of MessageWriter).  This should do the same.  Also, from a cosmetic perspective be sure to make the format of the debug and error messages consistent.  The current the debug message uses "queue {}" whereas the error message uses "queue:{}".

Reply:
Confirmed that exception handling has been moved to MessageWriter class and original message reinstated.